### PR TITLE
chore: upgrade to yarn 4

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Background
- Time to finally upgrade yarn.
#### Change List
- Upgrade to yarn 4
- Add minor missing dependencies etc.
- Add volta to github cli (volta is already being used and lets us control yarn and node version from package.json)
- Refresh yarn.lock
- Remove empty postinstall script for now as it caused problems for yarn.

One issue remains

- Fails because `yarn global` is no longer allowd, and ocular uses `yarn global add puppeteer`
- https://github.com/visgl/luma.gl/actions/runs/7059943383/job/19218524410?pr=1857
```
Usage Error: The 'yarn global' commands have been removed in 2.x - consider using 'yarn dlx' or a third-party plugin instead
```

### CODE REVIEW

**Comment by ibgreen** [comment]
- Fails because `yarn global` is no longer allowd, and ocular uses `yarn global add puppeteer`
- https://github.com/visgl/luma.gl/actions/runs/7059943383/job/19218524410?pr=1857
```
Usage Error: The 'yarn global' commands have been removed in 2.x - consider using 'yarn dlx' or a third-party plugin instead
```

File: .yarn/releases/yarn-4.0.2.cjs
Comment (ilijapuaca) [inline .yarn/releases/yarn-4.0.2.cjs:1]: Note that this checks in yarn binary into the repo, which somewhat beats the purpose of using volta, so you probably want to go with one or the other

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4.1.1
-      
+      - uses: volta-cli/action@v4
```
Comment (ilijapuaca) [inline .github/workflows/test.yml:19]: One issue that we've been seeing sporadically with this approach, is that it'd sometimes fail saying that there was an exceeding number of requests coming from a Github worker IP address. I assume that this is somewhat luck based as the worker pool is shared with other users

File: .yarnrc.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    - x64
+    - arm64
+    - darwin-x64
+  os:
+    - current
+    - darwin
+    - linux
+    - win32
+
+yarnPath: .yarn/releases/yarn-4.0.2.cjs
```
Comment (ilijapuaca) [inline .yarnrc.yml:14]: Per above, this may not be what you want to use

File: .github/workflows/test.yml (hunk 1 above)
Comment (ilijapuaca) [reply .github/workflows/test.yml:19]: Yes, just noting that it's something that's been happening on practice. One potential improvement that we wanted to make is caching the volta-managed versions using the `cache` action, which would avoid having anything checked in as well

**Comment by ibgreen** [comment]
Superseded by #1943

**Pull request closed without merging**